src/jobsautoreport/report.py

```diff
@@ -73,52 +120,49 @@ def __init__(self, querier: Querier):
     def _get_job_triggers_count(job_name: str, jobs: list[JobDetails]) -> int:
         return sum(1 for job in jobs if job.name == job_name)
 
-    def _get_job_states_count(
-        self, job_name: str, jobs: list[JobDetails]
-    ) -> JobStatesCount:
-        job_by_name = [job for job in jobs if job.name == job_name]
-        return self._get_states_count(job_by_name)
-
     @staticmethod
     def _get_number_of_jobs_by_state(
         jobs: list[JobDetails], job_state: JobState
     ) -> int:
         return len([job for job in jobs if job.state == job_state.value])
 
-    def _get_states_count(self, jobs: list[JobDetails]) -> JobStatesCount:
+    def _get_job_metrics(self, job_name: str, jobs: list[JobDetails]) -> JobMetrics:
+        job_by_name = [job for job in jobs if job.name == job_name]
+        return self._compute_job_metrics(job_by_name)
+
+    def _compute_job_metrics(self, jobs: list[JobDetails]) -> JobMetrics:
         total_jobs_number = len(jobs)
         successful_jobs_number = self._get_number_of_jobs_by_state(
             jobs=jobs, job_state=JobState.SUCCESS
         )
-        return (
-            JobStatesCount(successes=0, failures=0, failure_rate=0.0)
-            if total_jobs_number == 0
-            else JobStatesCount(
-                successes=successful_jobs_number,
-                failures=total_jobs_number - successful_jobs_number,
-                failure_rate=float(
-                    round(
-                        (
-                            (total_jobs_number - successful_jobs_number)
-                            / total_jobs_number
-                        )
-                        * 100,
-                        2,
-                    )
-                ),
-            )
+        if total_jobs_number == 0:
+            return JobMetrics(successes=0, failures=0)
+        return JobMetrics(
+            successes=successful_jobs_number,
+            failures=total_jobs_number - successful_jobs_number,
         )
 
-    @staticmethod
     def _get_top_n_jobs(
+        self,
         jobs: list[JobDetails],
-        computation_func: Callable[[str, list[JobDetails]], Any],
         n: int,
         comparison_func: Callable,
-    ) -> list[tuple[str, Any]]:
-        distinct_jobs = {job.name for job in jobs}
+    ) -> list[IdentifiedJobMetrics]:
+        distinct_jobs = {
+            JobIdentifier(
+                name=job.name,
+                repository=job.refs.repo,
+                base_ref=job.refs.base_ref,
+                context=job.context,
+            )
```

Comment:
```suggestion
            JobIdentifier.createFromJobDetails(job)
```

In order to make the creation of `JobIdentifer` more concise, and then have something like:
````
class JobIdentifier:
...
    @classmethod
    def createFromJobDetails(cls, job: JobDetails) -> JobIdentifier
         return cls(name=job.name,
                repository=job.refs.repo,
                base_ref=job.refs.base_ref,
                context=job.context,
          )
```